src/components/tables/Pagination.tsx

```diff
@@ -1,19 +1,32 @@
 import { Button, Flex, Text } from "@chakra-ui/react";
-import React, { Dispatch, SetStateAction } from "react";
+import { useRouter } from "next/router";
+import React, { Dispatch, SetStateAction, useLayoutEffect } from "react";
 
 type IPagination = {
   pages: number;
   currentPage: number;
   setCurrentPage: Dispatch<SetStateAction<number>>;
 };
 const Pagination = ({ pages, currentPage, setCurrentPage }: IPagination) => {
+  const router = useRouter();
+  const handleNextPage = (page: number) => {
+    router.replace({
+      query: { ...router.query, pageNum: page },
```

Comment:
let's just call this page
```suggestion
      query: { ...router.query, page },
```

Reply:
okay